src/common/dumper_test.go

```diff
@@ -142,3 +142,123 @@ func TestDumper(t *testing.T) {
 	want := strings.Contains(string(dat), `(11,"11\"xx\"","",NULL,210.01,NULL)`)
 	assert.True(t, want)
 }
+
+func TestDumperAll(t *testing.T) {
+	log := xlog.NewStdLog(xlog.Level(xlog.INFO))
+	fakedbs := driver.NewTestHandler(log)
+	server, err := driver.MockMysqlServer(log, fakedbs)
+	assert.Nil(t, err)
+	defer server.Close()
+	address := server.Addr()
+
+	selectResult := &sqltypes.Result{
+		Fields: []*querypb.Field{
+			{
+				Name: "id",
+				Type: querypb.Type_INT32,
+			},
+			{
+				Name: "name",
+				Type: querypb.Type_VARCHAR,
+			},
+			{
+				Name: "namei1",
+				Type: querypb.Type_VARCHAR,
+			},
+			{
+				Name: "null",
+				Type: querypb.Type_NULL_TYPE,
+			},
+			{
+				Name: "decimal",
+				Type: querypb.Type_DECIMAL,
+			},
+			{
+				Name: "datetime",
+				Type: querypb.Type_DATETIME,
+			},
+		},
+		Rows: make([][]sqltypes.Value, 0, 256)}
+
+	for i := 0; i < 201710; i++ {
+		row := []sqltypes.Value{
+			sqltypes.MakeTrusted(querypb.Type_INT32, []byte("11")),
+			sqltypes.MakeTrusted(querypb.Type_VARCHAR, []byte("11\"xx\"")),
+			sqltypes.MakeTrusted(querypb.Type_VARCHAR, []byte("")),
+			sqltypes.MakeTrusted(querypb.Type_NULL_TYPE, nil),
+			sqltypes.MakeTrusted(querypb.Type_DECIMAL, []byte("210.01")),
+			sqltypes.NULL,
+		}
+		selectResult.Rows = append(selectResult.Rows, row)
+	}
+
+	schemaResult := &sqltypes.Result{
+		Fields: []*querypb.Field{
+			{
+				Name: "Table",
+				Type: querypb.Type_VARCHAR,
+			},
+			{
+				Name: "Create Table",
+				Type: querypb.Type_VARCHAR,
+			},
+		},
+		Rows: [][]sqltypes.Value{
+			{
+				sqltypes.MakeTrusted(querypb.Type_VARCHAR, []byte("t1")),
+				sqltypes.MakeTrusted(querypb.Type_VARCHAR, []byte("CREATE TABLE `t1-05-11` (`a` int(11) DEFAULT NULL,`b` varchar(100) DEFAULT NULL) ENGINE=InnoDB")),
+			},
+		}}
+
+	tablesResult := &sqltypes.Result{
+		Fields: []*querypb.Field{
+			{
+				Name: "Tables_in_test",
+				Type: querypb.Type_VARCHAR,
+			},
+		},
+		Rows: [][]sqltypes.Value{
+			{
+				sqltypes.MakeTrusted(querypb.Type_VARCHAR, []byte("t1-05-11")),
+			},
+			{
+				sqltypes.MakeTrusted(querypb.Type_VARCHAR, []byte("t2-05-11")),
+			},
+		}}
+
+	// fakedbs.
+	{
+		fakedbs.AddQueryPattern("use .*", &sqltypes.Result{})
```

Comment:
Test errors:
 2020/03/16 12:17:09.147700 server.go:240:    	 [ERROR] 	server.handle.query.from.session[1].error:mock.handler.query[show databases].error[can.not.found.the.cond.please.set.first].query[SHOW DATABASES]

Here, we should add:
```
fakedbs.AddQueryPattern("show databases", &sqltypes.Result{?...})
```

Reply:
`{?...}` gives a build error and `{}` seems to work, but it seems like the result of `SHOW DATABASES` is empty as he dumps nothing => `dumping.all.done.cost[0.00sec].allrows[0].allbytes[0].rate[0.00MB/s]`

Reply:
Hmm, you should fill the values replace the ?... to mock the 'show databases' result.
&sqltypes.Result{} is empty.
Here is an example to mock `show tables from xx` result:
https://github.com/xelabs/go-mydumper/blob/master/src/common/dumper_test.go#L92

Reply:
Ok, I thought I got it. But I seem to have a minor problem with
```
fakedbs.AddQueryPattern("select * from `test1`.*", selectResult1)
```